compiler/modules/front/resolveTypes.bal

```diff
@@ -329,6 +338,18 @@ function resolveTypeDesc(ModuleSymbols mod, s:ModuleLevelDefn modDefn, int depth
     panic error("unimplemented type-descriptor");
 }
 
+function nonEmptyType(ModuleSymbols mod, s:ModuleLevelDefn modDefn, s:TypeDesc td, t:SemType ty) returns t:SemType|ResolveTypeError {
+    final t:Env env = mod.tc.env;
+    d:Location loc = s:locationInDefn(modDefn, { startPos: td.startPos, endPos: td.endPos });
```

Comment:
It bothers me that we are creating a loc here, but most of the time we won't be using it.

Reply:
Fixed